pkg/planner/core/point_get_plan.go

```diff
@@ -1157,6 +1160,64 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt, check bool
 	return nil
 }
 
+func checkGlobalIndexForPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt, pairs []nameValuePair, check bool) *PointGetPlan {
+	tblName, tblAlias := getSingleTableNameAndAlias(selStmt.From)
+	if tblName == nil {
+		return nil
+	}
+	tbl := tblName.TableInfo
+	if tbl == nil {
+		return nil
+	}
+
+	schema, names := buildSchemaFromFields(tblName.Schema, tbl, tblAlias, selStmt.Fields.Fields)
+	if schema == nil {
+		return nil
+	}
+	dbName := tblName.Schema.L
+	if dbName == "" {
+		dbName = ctx.GetSessionVars().CurrentDB
+	}
+
+	check = check || ctx.GetSessionVars().IsIsolation(ast.ReadCommitted)
+	check = check && ctx.GetSessionVars().ConnectionID > 0
+	var latestIndexes map[int64]*model.IndexInfo
+	var err error
+
+	for _, idxInfo := range tbl.Indices {
+		if !idxInfo.Unique || idxInfo.State != model.StatePublic || idxInfo.Invisible || idxInfo.MVIndex ||
+			!indexIsAvailableByHints(idxInfo, tblName.IndexHints) {
+			continue
+		}
+		if !idxInfo.Global {
+			continue
+		}
+		idxValues, idxConstant, colsFieldType := getIndexValues(idxInfo, pairs)
+		if idxValues == nil {
+			continue
+		}
+		if check && latestIndexes == nil {
+			latestIndexes, check, err = getLatestIndexInfo(ctx, tbl.ID, 0)
+			if err != nil {
+				logutil.BgLogger().Warn("get information schema failed", zap.Error(err))
+				return nil
+			}
+		}
+		if check {
+			if latestIndex, ok := latestIndexes[idxInfo.ID]; !ok || latestIndex.State != model.StatePublic {
+				continue
+			}
+		}
+		p := newPointGetPlan(ctx, dbName, schema, tbl, names)
```

Comment:
Should we also handle the explicit partition selection somewhere? Like `SELECT * FROM t partition(p1) where unique_key = 3` (see tblName.PartitionNames)

Reply:
Thanks for your remind! 
I will check the tblName.PartitionNames, the function will return nil if len(tblName.PartitionNames) > 0.

Reply:
I check it in this method.